frontend/packages/erd-core/src/components/ERDRenderer/convertDBStructureToNodes.ts

```diff
@@ -16,13 +16,30 @@ export const convertDBStructureToNodes = ({
 } => {
   const tables = Object.values(dbStructure.tables)
   const relationships = Object.values(dbStructure.relationships)
+  const sourceColumns = new Map<string, string>()
+  const tableColumnCardinalities = new Map<
+    string,
+    Record<string, Cardinality>
+  >()
+  for (const relationship of relationships) {
+    sourceColumns.set(
+      relationship.primaryTableName,
+      relationship.primaryColumnName,
+    )
+    tableColumnCardinalities.set(relationship.foreignTableName, {
+      ...tableColumnCardinalities.get(relationship.foreignTableName),
+      [relationship.foreignColumnName]: relationship.cardinality,
+    })
+  }
```

Comment:
I do believe this is a good change (movement). If my memory serves me right, `convertDBStructureToNodes` is always called, though it happens infrequently.

Reply:
Yes, it is called once before rendering the React Flow!